### ISSUE TITLE
`Format-Table` | Property's `PSEdition` values not exhibited

### ISSUE THREAD

**Issue by Ricky-Tigg** [PowerShell/PowerShell#20179]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://learn.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

```powershell
> Get-Module -ListAvailable | Where-Object -Property Name -Match "trusted" | Format-Table -AutoSize -Property ModuleType,Version,Name,PSEdition,ExportedCommands
```

### Expected behavior

```console
Exhibited in output:
- `Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules`
- Property's _PSEdition_ values, `Core,Desk `
as follows:


> Get-Module -ListAvailable | Where-Object -Property Name -Match "trusted" | Format-Table -AutoSize -Property ModuleType,Version,Name,PSEdition,ExportedCommands

    Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules
ModuleType Version Name                  PSEdition ExportedCommands
---------- ------- ----                  --------- ----------------
Manifest   2.0.0.0 TrustedPlatformModule Core,Desk {[Get-Tpm, Get-Tpm]…
```

### Actual behavior

```console
Not exhibited in output:
- `Directory: C:\WINDOWS\system32\WindowsPowerShell\v1.0\Modules`
- Property's _PSEdition_ values, `Core,Desk `
as follows:


> Get-Module -ListAvailable | Where-Object -Property Name -Match "trusted" | Format-Table -AutoSize -Property ModuleType,Version,Name,PSEdition,ExportedCommands

ModuleType Version Name                  PSEdition ExportedCommands
---------- ------- ----                  --------- ----------------
  Manifest 2.0.0.0 TrustedPlatformModule           {[Get-Tpm, Get-Tpm]…
```

### Error details

```console
_
```

### Environment data

```powershell
> PSVersion                      7.3.6
PSEdition                      Core
GitCommitId                    7.3.6
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_

**Comment by mklement0** [comment]
It sounds like you're looking for the `.CompatiblePSEditions` property instead.

**Comment by Ricky-Tigg** [comment]
The _PSEdition_-named property as an exception among the properties exhibited by the `Get-Module -ListAvailable`'s output, doesn't query the values that it reports while being as property of `Format-Table` pipe-line. Your mention _CompatiblePSEditions_, which is not even exhibited by the `Get-Module -ListAvailable`'s output, cannot be logically.deduced. If it was not intuition or inspiration, then i have no clue what led you to determine that it was the option to apply.
```powershell
> Get-Module -ListAvailable | Where-Object -Property Name -Match "trusted" | Format-Table -Property CompatiblePSEditions
CompatiblePSEditions
--------------------
{Desktop, Core}
```
If i had had to seek for something, that would have been most likely consistency.

**Comment by mklement0** [comment]
The behavior is provided by the _experimental_ [PSCustomTableHeaderLabelDecoration](https://learn.microsoft.com/en-us/powershell/scripting/learn/experimental-features#pscustomtableheaderlabeldecoration) feature (available starting with preview versions of v7.4), which applies special formatting - italics by default, controllable via `$PSStyle.Formatting.CustomTableHeaderLabel` - to any column name that doesn't match the name of a property of the type being formatted.

As with any [experimental feature](https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_Experimental_Features), it isn't _guaranteed_ to become a stable feature, however.

**Comment by Ricky-Tigg** [comment]
As the column named _PSEdition_ of the `Get-Module -ListAvailable`'s output on purpose differs from the underlying property name _CompatiblePSEditions_, and the `Get-Module -ListAvailable | Format-Table -Property PSEdition`'s output is correct, so there is no issue.
